tests/test_views/test_search.py

```diff
@@ -0,0 +1,777 @@
+            result = response.json()
+            result_ids.extend([r["id"] for r in result["collection"]])
+
+        expected_ids = threads_ids[::-1]
+        assert result_ids == expected_ids
+
+    check_pagination(1, 50)
+    check_pagination(30, 2)
+    check_pagination(None, 3)
+
+
+def test_sorting(api_client: APIClient) -> None:
+    """
+    Test the sorting functionality for threads based on various criteria, such as date, activity, votes, and comments.
+    Asserts that the threads are sorted correctly according to the specified sorting key.
+    """
+    course_id = "course-v1:Arbisoft+SE002+2024_S2"
+
+    # Create and save threads
+    threads_ids = []
+    for i in range(6):
+        thread = CommentThread().insert(
+            title=f"title-{i}",
+            body="text",
+            author_id="1",
+            course_id=course_id,
+            commentable_id="dummy",
+        )
+        threads_ids.append(thread)
+        # Add a slight delay to ensure created_date is different
+        time.sleep(0.001)
+
+    # Update specific threads to simulate activity, votes, and comments
+    votes = CommentThread().get_votes_dict(up=["1"], down=[])
+    CommentThread().update(thread_id=threads_ids[1], votes=votes)
+    CommentThread().update(thread_id=threads_ids[2], votes=votes)
+    CommentThread().update(thread_id=threads_ids[1], comments_count=5)
+    CommentThread().update(thread_id=threads_ids[3], comments_count=5)
+
+    refresh_elastic_search_indices()
+
+    def fetch_and_check(sort_key: Optional[str], expected_indexes: list[int]) -> None:
+        params = {"text": "text"}
+        if sort_key:
+            params["sort_key"] = str(sort_key)
+
+        query_string = urllib.parse.urlencode(params)
+        response = get_search_response(api_client, query_string, threads_ids)
+        assert_result_total(response, 6)
+        result = response.json()
+        threads = result["collection"]
+        expected_ids = [threads_ids[i] for i in expected_indexes]
+        actual_ids = [thread["id"] for thread in threads]
+        assert (
+            actual_ids == expected_ids
+        ), f"Expected {expected_ids}, but got {actual_ids}"
+
+    # Test various sorting scenarios
+    fetch_and_check("date", [5, 4, 3, 2, 1, 0])
+    fetch_and_check("activity", [5, 4, 3, 2, 1, 0])
+    fetch_and_check("votes", [2, 1, 5, 4, 3, 0])
+    fetch_and_check("comments", [3, 1, 5, 4, 2, 0])
+    fetch_and_check(None, [5, 4, 3, 2, 1, 0])  # Default sorting by date
+
+
+def test_spelling_correction(api_client: APIClient) -> None:
+    """
+    Test the spelling correction feature in search.
+    Verifies that misspelled words in both thread titles and comment bodies are correct
+    """
+    commentable_id = "test_commentable"
+    thread_title = "a thread about green artichokes"
+    comment_body = "a comment about greed pineapples"
+
+    thread_id = CommentThread().insert(
+        title=thread_title,
+        body="text",
```

Comment:
Setting `body=thread_title` is more consistent with v1 and resolves the `check_correction("pinapples", "pineapples")` below. This is because, according to the phrase suggester, "pinapples" matches "text" with a confidence of 0.7237521 (don't ask me why). But "pinapples" matches "pineapples" with a confidence of just 0.47230867. Because 0.47230867 < 0.7237521, "pineapples" is not returned:

> a confidence level of 1.0 will only return suggestions that score higher than the input phrase.

([source](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html#_basic_phrase_suggest_api_parameters))